src/main/sensors/gyroanalyse.c

```diff
@@ -45,19 +45,26 @@
 // A sampling frequency of 1000 and max frequency of 500 at a window size of 32 gives 16 frequency bins each with a width 31.25Hz
 // Eg [0,31), [31,62), [62, 93) etc
 
-#define FFT_BIN_COUNT             (FFT_WINDOW_SIZE / 2)
-// for gyro loop >= 4KHz, analyse up to 666Hz, 16 bins each 41.625 Hz wide
-#define FFT_SAMPLING_RATE_HZ      1333
+// for gyro loop >= 4KHz, analyse up to 1000Hz, 16 bins each 62.5 Hz wide
+#define FFT_SAMPLING_RATE_HZ      2000
+#define FFT_RESOLUTION            ((float)FFT_SAMPLING_RATE_HZ / FFT_WINDOW_SIZE)
 // following bin must be at least 2 times previous to indicate start of peak
-#define FFT_MIN_BIN_RISE          2
+#define FFT_BIN_COUNT             (FFT_WINDOW_SIZE / 2)
 // the desired approimate lower frequency when calculating bin offset
 #define FFT_BIN_OFFSET_DESIRED_HZ 90
-// lowpass frequency for smoothing notch centre point
-#define DYN_NOTCH_SMOOTH_FREQ_HZ  60
+// centre frequency of bandpass that constrains input to FFT
+#if FFT_SAMPLING_RATE_HZ == 2000
+#define FFT_BPF_HZ                350
+#elif FFT_SAMPLING_RATE_HZ < 2000
+#define FFT_BPF_HZ                (FFT_SAMPLING_RATE_HZ / 4)
+#endif
+#define DYN_NOTCH_SMOOTH_FREQ_HZ  50
 // notch centre point will not go below this, must be greater than cutoff, mid of bottom bin
-#define DYN_NOTCH_MIN_CENTRE_HZ   125
+#define DYN_NOTCH_MIN_CENTRE_HZ   140
+// maximum notch centre frequency limited by Nyquist
+#define DYN_NOTCH_MAX_CENTRE_HZ   (FFT_SAMPLING_RATE_HZ / 2)
 // lowest allowed notch cutoff frequency
-#define DYN_NOTCH_MIN_CUTOFF_HZ   105
+#define DYN_NOTCH_MIN_CUTOFF_HZ   110
```

Comment:
When changes like this sneak into refactoring they often go unnoticed. Better change defaults in a separate PR.

Reply:
These were all based on Q of 0.7, and will all change when the ability to adjust the main frequency is implemented.  Since the way the filter operates is now very different, and will be intended to operate differently I think they will inevitably change.  The right value for a micro is different from a 5" or a 6".